src/Raven.Studio/typescript/viewmodels/manage/trafficWatch.ts

```diff
@@ -13,33 +13,73 @@ import virtualColumn = require("widgets/virtualGrid/columns/virtualColumn");
 import recentQueriesStorage = require("common/storage/savedQueriesStorage");
 import queryCriteria = require("models/database/query/queryCriteria");
 import databasesManager = require("common/shell/databasesManager");
+import TrafficWatchHttpChange = Raven.Client.Documents.Changes.TrafficWatchHttpChange;
+import trafficWatchQueriesDialog from "viewmodels/manage/trafficWatchQueriesDialog";
+import app = require("durandal/app");
 
 type trafficChangeType = Raven.Client.Documents.Changes.TrafficWatchChangeType | Raven.Client.ServerWide.Tcp.TcpConnectionHeaderMessage.OperationTypes; 
 
 class runQueryFeature implements columnPreviewFeature {
-    install($tooltip: JQuery, valueProvider: () => any, elementProvider: () => any, containerSelector: string) {
+    install($tooltip: JQuery, valueProvider: () => any, elementProvider: () => any, containerSelector: string): void {
         $tooltip.on("click", ".run-query", () => {
-            const value = valueProvider();
-
+            let value = valueProvider();
             const item: Raven.Client.Documents.Changes.TrafficWatchChangeBase = elementProvider();
-            
-            const query = queryCriteria.empty();
 
-            query.queryText(value);
-            query.name("Traffic watch query"); 
-            query.recentQuery(true);
+            if (item.TrafficWatchType !== "Http" || (item as TrafficWatchHttpChange).Type !== "MultiGet") {
+                runQueryFeature.executeQuery(value, item);
+                return;
+            }
+
+            const queryList = runQueryFeature.getQueryList(value);
+            if (queryList.length === 1) {
+                runQueryFeature.executeQuery(queryList[0], item);
+            } else {
+                app.showBootstrapDialog(new trafficWatchQueriesDialog(queryList))
+                    .done(queryToExecute => {
+                        if (queryToExecute) {
+                            runQueryFeature.executeQuery(queryToExecute, item);
+                        }
+                    });
+            }
+        });
+    }
 
-            const queryDto = query.toStorageDto();
+    private static getQueryList(value: any): string[] {
+        const queryList: string[] = [];
 
-            const db = databasesManager.default.getDatabaseByName(item.DatabaseName);
-            
-            recentQueriesStorage.saveAndNavigate(db, queryDto, { newWindow: true });
+        const lines = value.split('\r\n');
```

Comment:
what if we have line breaks inside query? Do we handle that correctly?

Reply:
IMHO - better to see the query as was used for debugging purposes - isn't it ?

Reply:
it is fine - no need to change